tests/unit_tests/external_interaction/nexus/test_compare_nexus_to_gda_exhaustively.py

```diff
@@ -0,0 +1,266 @@
+import shutil
+from pathlib import Path
+from unittest.mock import MagicMock
+
+import h5py
+import pytest
+from h5py import Dataset, Datatype, File, Group
+from numpy import dtype
+
+from ....conftest import extract_metafile
+
+TEST_DATA_DIRECTORY = Path("tests/test_data/nexus_files/rotation")
+TEST_EXAMPLE_NEXUS_FILE = Path("ins_8_5.nxs")
+TEST_NEXUS_FILENAME = "rotation_scan_test_nexus_0.nxs"
+TEST_METAFILE = "ins_8_5_meta.h5.gz"
+FAKE_DATAFILE = "fake_data.h5"
+
+h5item = Group | Dataset | File | Datatype
+
+
+def get_groups(dataset: h5py.File) -> set:
+    e = set()
+
+    def add_layer(s: set, d: h5item):
+        if isinstance(d, h5py.Group):
+            for k in d:
+                s.add(d.name)
+                add_layer(s, d[k])
+
+    add_layer(e, dataset)
+    return e
+
+
+def has_equiv_in(item: str, groups: set, exception_table: dict[str, set[str]]):
+    if item not in exception_table.keys():
+        return False
+    # one of the items in exception_table[item] must be in the tested groups
+    return exception_table[item] & groups != set()
```

Comment:
wouldn't it be clearer just to do `not exception_table[item] in groups` ?

Reply:
the equivalent would be something like `any([ex in groups for ex in exception_table[item]])`, it's testing that at least one item (but possibly more) in the set in the exception table is also in the set of groups